allennlp/data/tokenizers/basic_lm.py

```diff
@@ -0,0 +1,29 @@
+from typing import List
+
+from overrides import overrides
+
+from allennlp.common import Params
+from allennlp.data.tokenizers.token import Token
+from allennlp.data.tokenizers.tokenizer import Tokenizer
+
+@Tokenizer.register("basic_lm")
+class BasicTokenizer(Tokenizer):
```

Comment:
I feel like it's confusing if the registered name doesn't agree with the class name, so that either the registered name (and the filename) should be "basic" or the classname should be `BasicLmTokenizer`, your call on which is more appropriate

Reply:
also, can you add a docstring explaining what this tokenizer does and why someone would choose to use it